### ISSUE TITLE
[java] DontCallThreadRun reported for java.lang.Thread subclass where overridden run() calls super.run()

### ISSUE THREAD

**Issue by slovdahl** [pmd/pmd#5439]
**Affects PMD Version:** 7.9.0

**Rule:** DontCallThreadRun

Please provide the rule name and a link to the rule documentation: https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#dontcallthreadrun

**Description:** A `java.lang.Thread` subclass that overrides `Thread#run` and calls `super.run()` gets reported by the `DontCallThreadRun` rule.

**Code Sample demonstrating the issue:**

```java
package test;

public class TestThread extends Thread {

    public TestThread(Runnable task) {
        super(task);
    }

    @Override
    public void run() {
        super.run();
    }
}
```

**Expected outcome:**

PMD reports a violation at line 11, but that's wrong. That's a false positive. I think it's a valid case of calling `Thread#run`, while still wanting to catch calls to `Thread#run` outside subclassed `Thread#run` methods.

```
TestThread.java:11:        DontCallThreadRun:      Dont call Thread.run() explicitly, use Thread.start()
```

**Running PMD through:** *Gradle*

**Comment by zbynek** [comment]
Closing as duplicate of #6528 (should be the other way around, but that one already has a PR)
